### ISSUE TITLE
Failure to load kernel modules in 6.4.7-surface #3

### ISSUE THREAD

**Issue by j-gama-oliveira** [linux-surface/linux-surface#1219]
After update to the latest linux-surface kernel (6.4.7-surface #3) in my SP6, camera stopped working (I had previously implemented the GStreamer Loopback Device, so that I would start the camera manually from the command line). I think the problem comes from failure to load kernel modules at boot time, since I get a message «[FAILED] Failed to start Load Kernel Modules», which didn't show up before update to the lastest kernel.
Thanks for any help on this, and for the great job so far.

### Environment

- Hardware model: Surface Pro 6
- Kernel version: Linux surface 6.4.7-surface #3 SMP PREEMPT_DYNAMIC Tue Aug  1 21:23:55 UTC 2023 x86_64 x86_64 x86_64 GNU/Linux
- Distribution: Ubuntu

<details><summary>`dmesg` output</summary>
[dmesg.out.txt](https://github.com/linux-surface/linux-surface/files/12327189/dmesg.out.txt)
</details>

**Comment by qzed** [comment]
There are a bunch of the following lines in the log:
```
[    2.567912] Loading of unsigned module is rejected
```

That looks like secureboot prevents you from loading some modules and could explain the message you see.

What happens if you try to load the loopback module manually?

**Comment by j-gama-oliveira** [comment]
That might be it then (secureboot preventing loading some modules), though I have had the camera working for a couple of months now and have been keeping the linux-surface kernel always updated, and had no such problem with any of the updates up to the latest.

I just tried to run

sudo modprobe v4l2loopback video_nr=42 card_label="virtualcam" exclusive_caps=1

and indeed it got rejected:

modprobe: ERROR: could not insert 'v4l2loopback': Key was rejected by service

**Comment by qzed** [comment]
Right, that sounds like secureboot. You can try disabling it and see if that works. In the long term, you probably want to try and set up secureboot support / auto-signing for those modules.

**Comment by j-gama-oliveira** [comment]
Still didn't have the opportunity to do it, but will eventually. Thanks!

Maximilian Luz ***@***.***> escreveu no dia domingo,
27/08/2023 à(s) 19:30:

**Comment by hax0rbana-adam** [comment]
I ran into this same problem with the virtualbox kernel module.

The core issue is that neither `CONFIG_MODULE_SIG_FORCE=y` nor `CONFIG_LOCK_DOWN_IN_EFI_SECURE_BOOT=y` is set in the kernel config. Adding either or these to `/lib/modules/6.13.6-surface-1/build/.config` will cause  `vboxconfig` to sign the kernel modules which it compiles.

After that, all the `Loading of unsigned module is rejected` errors go away! ❤ 

If we can get either of these options added to the kernel config, it would reduce the errors about signed kernel modules.

**Comment by hax0rbana-adam** [comment]
BTW, I just realized `/lib/modules/6.13.6-surface-1/build/` is a symlink to `/usr/src/linux-headers-6.13.6-surface-1`, so this would be a change to the `linux-headers` package. I think setting `CONFIG_LOCK_DOWN_IN_EFI_SECURE_BOOT=y` would make the most sense.
